src/particles/particle_spin_pol.jl

```diff
@@ -0,0 +1,130 @@
+"""
+Abstract base type for the spin or polarization of [`FermionLike`](@ref) or [`BosonLike`](@ref) particles, respectively.
+"""
+abstract type AbstractSpinOrPolarization end
+
+"""
+Abstract base type for the spin of [`FermionLike`](@ref) particles.
+"""
+abstract type AbstractSpin <: AbstractSpinOrPolarization end
+
+"""
+Abstract base type for definite spins of [`FermionLike`](@ref) particles.
+
+Concrete types are [`SpinUp`](@ref) and [`SpinDown`](@ref).
+"""
+abstract type AbstractDefiniteSpin <: AbstractSpin end
+
+"""
+Abstract base type for indefinite spins of [`FermionLike`](@ref) particles.
+
+One concrete type is [`AllSpin`](@ref).
+"""
+abstract type AbstractIndefiniteSpin <: AbstractSpin end
+
+"""
+Concrete type indicating that a [`FermionLike`](@ref) has spin-up.
+"""
+struct SpinUp <: AbstractDefiniteSpin end
+
+"""
+Concrete type indicating that a [`FermionLike`](@ref) has spin-down.
+"""
+struct SpinDown <: AbstractDefiniteSpin end
+
+"""
+Concrete type indicating that a [`FermionLike`](@ref) has an indefinite spin and should average or sum over all spins, depending on direction.
```

Comment:
What do you mean by "depending on direction"?

Reply:
If I recall correctly what Uwe said was that for Incoming particles, the  result will be averaged over the possible outcomes, and for outgoing particles it will be summed.

Reply:
Right, potentially, over spins/polarisations incoming particles will be averaged, but over outgoing particles will just be summed.

Reply:
I elaborated the doc a little, if it's clearer now please resolve the conversation.

Reply:
I see, for some reason I thought of x/y/z instead of incoming/outgoing as direction. The edit makes it clear.